ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt

```diff
@@ -0,0 +1,60 @@
+package com.chartboost.mediation.referenceadapter.sdk
+
+import android.content.Context
+import java.lang.ref.WeakReference
+
+class ReferenceSettings private constructor(context: Context) {
```

Comment:
why not just make this an object? and have get/set take context? should make this simpler to understand and not have to use synchronized.